packages/driver/test/cypress/integration/commands/xhr_spec.coffee

```diff
@@ -69,6 +69,26 @@ describe "src/cy/commands/xhr", ->
           expect(onreadystatechanged).to.be.true
           expect(xhr.status).to.eq(404)
 
+    it "allow multiple readystatechange calls", ->
+      responseText = null
+      responseStatuses = 0
+
+      cy
+        .server()
+        .route({url: /longtext.txt/}).as("getFoo")
+        .window().then (win) ->
+          xhr = new win.XMLHttpRequest
+          xhr.onreadystatechange = ->
+            responseText = xhr.responseText
+            if xhr.readyState == 3
+              responseStatuses++
+          xhr.open("GET", "/fixtures/longtext.txt?" + (new Date).getTime())
```

Comment:
you could just do `Cypress._.random(...)` here, it would better reflect what you are trying to do